test/parallel/test-async-local-storage-exit-does-not-leak.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const { AsyncLocalStorage } = require('async_hooks');
+
+const als = new AsyncLocalStorage();
+
+// The als instance should be getting removed from the storageList in
+// lib/async_hooks.js when exit(...) is called, therefore when the nested runs
+// are called there should be no copy of the als in the storageList to run the
+// _propagate method on.
+als._propagate = common.mustNotCall(als._propagate);
+
+const done = common.mustCall();
+
+function run(count) {
+  if (count === 0) return done();
+  als.run({}, () => {
+    als.exit(run, --count);
```

Comment:
Maybe add asserts that store is set just before `als.exit()` and not set just before `als.run()`?

Reply:
Not sure that's relevant as that's true before the change too. It's not that it has a value to propagate, it's just that it still thinks it should be trying to propagate a context object into it, even though it would be nothing at the time.

Reply:
I think we already validate this in other tests, as it's a part of basic ALS functionality. See `test/async-hooks/test-async-local-storage-nested.js` for instance.